### PR TITLE
Dockerized ISS SCMS Health Checker

### PR DESCRIPTION
**Pull request by drewjj**
This pull request includes the fully dockerized ISS SCMS Health Checker that was originally a cloud function in the private CDOT repository.
- Full dockerized scripts and Docker related files
- README to explain how to setup the image and describe requirements and limitations
- Unit tests
- Small update to other README to match details added to environment variables

### CODE REVIEW

File: addons/images/iss_health_check/iss_token.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+from google.cloud import secretmanager
+import requests
+import os
+import json
+import uuid
+import logging
+
+def create_secret(client, secret_id, parent):
```
Comment (dmccoystephenson) [inline addons/images/iss_health_check/iss_token.py:8]: It's not immediately clear what the purpose of the parameters of these methods are when reading the method signatures. We should consider adding method descriptor comments or specifying types to make this clearer.

File: addons/tests/iss_health_check/test_iss_health_cheker.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
+from unittest.mock import patch, MagicMock, Mock
```
Comment (dmccoystephenson) [inline addons/tests/iss_health_check/test_iss_health_cheker.py]: It looks like this filename has a typo, should it be called `test_iss_health_checker.py`?

File: addons/tests/iss_health_check/test_iss_token.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+  }
+  mock_sm_client.create_secret.assert_called_with(request=expected_request)
+
+@patch('images.iss_health_check.iss_token.secretmanager.SecretManagerServiceClient')
+@patch('images.iss_health_check.iss_token.secretmanager')
+def test_check_if_secret_exists_true(mock_secretmanager, mock_sm_client):
+  mock_secretmanager.ListSecretsRequest.return_value = "list-request"
+
+  item_match = MagicMock()
+  item_match.name = "proj/test-proj/sercret/test-secret_id"
```
Comment (dmccoystephenson) [inline addons/tests/iss_health_check/test_iss_token.py]: Is `sercret` here a typo?

File: addons/tests/iss_health_check/test_iss_token.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  mock_sm_client.list_secrets.assert_called_with(request="list-request")
+  assert actual_value == True
+
+@patch('images.iss_health_check.iss_token.secretmanager.SecretManagerServiceClient')
+@patch('images.iss_health_check.iss_token.secretmanager')
+def test_check_if_secret_exists_false(mock_secretmanager, mock_sm_client):
+  mock_secretmanager.ListSecretsRequest.return_value = "list-request"
+
+  item_not_match = MagicMock()
+  item_not_match.name = "proj/test-proj/sercret/test-secret"
```
Comment (dmccoystephenson) [inline addons/tests/iss_health_check/test_iss_token.py]: Is `sercret` here a typo?

File: addons/images/iss_health_check/README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# ISS Health Checker
```
Comment (dmccoystephenson) [inline addons/images/iss_health_check/README.md]: We should write out what `ISS` stands for somewhere in this document.

File: addons/images/iss_health_check/README.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+## Table of Contents
+
+- [ISS Health Checker](#iss-health-checker)
+  - [Table of Contents](#table-of-contents)
+  - [About ](#about-)
+  - [Requirements ](#requirements-)
+
+## About <a name = "about"></a>
+
+This directory contains a microservice that runs within the CV Manager GKE Cluster. The iss_health_checker application populates the CV Manager PostGreSQL database's 'scms_health' table with the current ISS SCMS statuses of all RSUs recorded in the 'rsus' table. These statuses are queried by this application from a provided ISS Green Hills SCMS API endpoint.
```
Comment (dmccoystephenson) [inline addons/images/iss_health_check/README.md:12]: We should write out what SCMS stands for as well somewhere in this document.

**Review by dmccoystephenson — APPROVED** [review]
The code looks good & the tests pass! Just a few minor comments/suggestions below.

File: addons/images/iss_health_check/iss_token.py (hunk 1 above)
Comment (drewjj) [reply addons/images/iss_health_check/iss_token.py:8]: Addressed

File: addons/tests/iss_health_check/test_iss_health_cheker.py (hunk 2 above)
Comment (drewjj) [reply addons/tests/iss_health_check/test_iss_health_cheker.py]: Addressed

File: addons/tests/iss_health_check/test_iss_token.py (hunk 3 above)
Comment (drewjj) [reply addons/tests/iss_health_check/test_iss_token.py]: Addressed

File: addons/tests/iss_health_check/test_iss_token.py (hunk 4 above)
Comment (drewjj) [reply addons/tests/iss_health_check/test_iss_token.py]: Addressed

File: addons/images/iss_health_check/README.md (hunk 5 above)
Comment (drewjj) [reply addons/images/iss_health_check/README.md]: Addressed

File: addons/images/iss_health_check/README.md (hunk 6 above)
Comment (drewjj) [reply addons/images/iss_health_check/README.md:12]: Addressed

**Review by dmccoystephenson — APPROVED** [review]
Looks good!

**Pull request merged**